nixos/modules/services/hardware/acpid.nix

```diff
@@ -3,153 +3,189 @@
 with lib;
 
 let
-  cfg = config.services.acpid;
-
-  canonicalHandlers = {
-    powerEvent = {
-      event = "button/power.*";
-      action = cfg.powerEventCommands;
-    };
-
-    lidEvent = {
-      event = "button/lid.*";
-      action = cfg.lidEventCommands;
-    };
-
-    acEvent = {
-      event = "ac_adapter.*";
-      action = cfg.acEventCommands;
-    };
-  };
-
-  acpiConfDir = pkgs.runCommand "acpi-events" { preferLocalBuild = true; }
-    ''
-      mkdir -p $out
-      ${
-        # Generate a configuration file for each event. (You can't have
-        # multiple events in one config file...)
-        let f = name: handler:
-          ''
-            fn=$out/${name}
-            echo "event=${handler.event}" > $fn
-            echo "action=${pkgs.writeShellScriptBin "${name}.sh" handler.action }/bin/${name}.sh '%e'" >> $fn
-          '';
-        in concatStringsSep "\n" (mapAttrsToList f (canonicalHandlers // cfg.handlers))
-      }
+  stage1Cfg = config.boot.initrd.services.acpid;
+  stage2Cfg = config.services.acpid;
```

Comment:
This might be confusing as initrd has stage 1 and 2, too.

Reply:
so would `initrdCfg` / `mainCfg` work? I'm not sure what to name the one under `config.services`.